javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -549,6 +562,34 @@ private void extractSource() throws IOException {
     }
   }
 
+  protected void installDependencies(Set<Path> filesToExtract) {
+    for (Path file : filesToExtract) {
+      if (file.getFileName().toString().equals("package.json")) {
+        System.out.println("Installing dependencies from " + file);
+        ProcessBuilder pb =
+            new ProcessBuilder(
+                Arrays.asList(
+                    "yarn",
```

Comment:
Do we want to check somewhere whether `yarn` is installed? I think on Ubuntu at least there is also some other package providing [a command](http://git.liw.fi/cgi-bin/cgit/cgit.cgi/cmdtest/tree/README.yarn) called `yarn` that has nothing to do with npm, which might cause very confusing errors.

Reply:
Do you have any ideas for how to distinguish the two? I mean we can run `yarn -v` to check that _something_ called `yarn` is installed, but the other `yarn` probably also supports that. We could run `yarn ---help` and match some part of the output but there's no guarantee it will keep working in future versions.

Reply:
The documentation of the other yarn doesn't look like it supports `-v`, but they may just think that it's an obvious option to support. And of course any other yarns out there probably also support `-v`. So just checking that something called yarn is installed is probably fine?